core/src/main/scala/agni/ColumnGetter.scala

```diff
@@ -100,54 +102,94 @@ trait LowPriorityColumnGetter {
-        row.getList(i, classOf[Any]).map(_.asInstanceOf[A]).toList
+  def streamColumnGetter[A, A0](f: A => A0)(implicit tag: ClassTag[A]): ColumnGetter[Stream[A0]] =
+    new ColumnGetter[Stream[A0]] {
+      def apply(row: Row, i: Int): Stream[A0] =
+        row.getList(i, TypeToken.of[A](tag.runtimeClass.asInstanceOf[Class[A]])).toStream.map(f)
     }
 
-  implicit def setColumnGetter[A](implicit tag: ClassTag[A]): ColumnGetter[Set[A]] =
-    new ColumnGetter[Set[A]] {
-      def apply(row: Row, i: Int): Set[A] =
-        row.getSet(i, classOf[Any]).map(_.asInstanceOf[A]).toSet
+  def seqColumnGetter[A, A0](f: A => A0)(implicit tag: ClassTag[A]): ColumnGetter[Seq[A0]] =
+    new ColumnGetter[Seq[A0]] {
+      def apply(row: Row, i: Int): Seq[A0] =
+        row.getList(i, TypeToken.of[A](tag.runtimeClass.asInstanceOf[Class[A]])).map(_.asInstanceOf[A0])
     }
 
-  implicit def mapColumnGetter[K, V](
-    implicit
-    keyTag: ClassTag[K],
-    valTag: ClassTag[V]
-  ): ColumnGetter[Map[K, V]] =
-    new ColumnGetter[Map[K, V]] {
-      def apply(row: Row, i: Int): Map[K, V] = {
-        val x = row.getMap(i, classOf[Any], classOf[Any]) map { case (k, v) => (k.asInstanceOf[K], v.asInstanceOf[V]) }
-        x.toMap
-      }
+  def vectorColumnGetter[A, A0](f: A => A0)(implicit tag: ClassTag[A]): ColumnGetter[Vector[A0]] =
+    new ColumnGetter[Vector[A0]] {
+      def apply(row: Row, i: Int): Vector[A0] =
+        row.getList(i, TypeToken.of[A](tag.runtimeClass.asInstanceOf[Class[A]])).toVector.map(_.asInstanceOf[A0])
     }
 
-  implicit val udtValueColumnGetter: ColumnGetter[UDTValue] =
-    new ColumnGetter[UDTValue] {
-      def apply(row: Row, i: Int): UDTValue = row.getUDTValue(i)
+  def listColumnGetter[A, A0](f: A => A0)(implicit tag: ClassTag[A]): ColumnGetter[List[A0]] =
+    new ColumnGetter[List[A0]] {
+      def apply(row: Row, i: Int): List[A0] =
+        row.getList(i, TypeToken.of[A](tag.runtimeClass.asInstanceOf[Class[A]])).toList.map(f)
     }
 
-  implicit val tupleValueColumnGetter: ColumnGetter[TupleValue] =
-    new ColumnGetter[TupleValue] {
-      def apply(row: Row, i: Int): TupleValue = row.getTupleValue(i)
+  def setColumnGetter[A, A0](f: A => A0)(implicit tag: ClassTag[A]): ColumnGetter[Set[A0]] =
+    new ColumnGetter[Set[A0]] {
+      def apply(row: Row, i: Int): Set[A0] = {
+        val x: Set[A] = row.getSet(i, TypeToken.of[A](tag.runtimeClass.asInstanceOf[Class[A]])).toSet
+        x.map(f)
+      }
     }
+
+  def mapColumnGetter[K, V, V0](f: V => V0)(
+    implicit
+    keyTag: ClassTag[K],
+    valTag: ClassTag[V]
+  ): ColumnGetter[Map[K, V0]] =
+    new ColumnGetter[Map[K, V0]] {
+      def apply(row: Row, i: Int): Map[K, V0] =
+        row.getMap(
+          i,
+          TypeToken.of[K](keyTag.runtimeClass.asInstanceOf[Class[K]]),
+          TypeToken.of[V](keyTag.runtimeClass.asInstanceOf[Class[V]])
+        ).toMap.mapValues(f)
+    }
+
+  implicit val setS = setColumnGetter[String, String](identity)
```

Comment:
`TypeToken.of[A]` is be failed to cast anyway, so we have to explicitly annotate.